pkg/api/routes.go

```diff
@@ -949,14 +949,6 @@ func (rh *RouteHandler) PatchBlobUpload(response http.ResponseWriter, request *h
 			return
 		}
 
-		contentRange := request.Header.Get("Content-Range")
```

Comment:
This part got deleted? why?

Reply:
Because on line 937 we have this condition
`if request.Header.Get("Content-Length") == "" || request.Header.Get("Content-Range") == ""`

since the removed code is on the else branch, testing for Content-Range to be empty will always result false, so it's unreachable.

Reply:
maybe there should be an AND instead of OR?

Reply:
I have tested this change and it breaks Zot.
`Summarizing 2 Failures:

[Fail] OCI Distribution Conformance Tests Push Blob Upload Streamed [It] PATCH request with blob in body should yield 202 response 
/go/src/github.com/opencontainers/distribution-spec/conformance/02_push_test.go:35

[Panic!] OCI Distribution Conformance Tests Push Blob Upload Streamed [It] PUT request to session URL with digest should yield 201 response 
/usr/local/go/src/runtime/panic.go:199
`

Reply:
I think the initial logic is good if it's compliant with the OCI tests. They seem quite exhaustive. I propose to just remove the redundant `if`.

Reply:
https://github.com/project-zot/zot/pull/527

Refactored like so ^. This is what this check really means.

Reply:
https://github.com/project-zot/zot/runs/6375017852?check_suite_focus=true
and conformance passes ^

Reply:
https://pkg.go.dev/net/http#Header.Set
^ Is it possible to set the key but empty value in this case?

Reply:
I am sorry but I don't understand what you mean by the last comment. I have written a small (but exhaustive) equivalent example to show that the line is unreachable.

```
package main

import (
	"fmt"
)

func foo(contentIsPresent, rangeIsPresent bool) {
	if !contentIsPresent || !rangeIsPresent { // equivalent to line 943
		fmt.Println("PutBlobChunkStreamed")
	} else {
		if !rangeIsPresent {                  // equivalent to line 959
			fmt.Println("Unreachable line")   
		}
	}
}

func main() {
	truthValues := [2]bool{true, false}

	for _, i := range truthValues {
		for _, j := range truthValues {
			foo(i, j)
		}
	}
}
```